mithril-test-lab/mithril-end-to-end/src/main.rs

```diff
@@ -198,9 +208,66 @@ async fn main() -> StdResult<()> {
 
     app_stopper.stop().await;
     join_set.shutdown().await;
+
     res
 }
 
+#[derive(Debug)]
+enum AppResult {
+    Success(),
+    UnretryableError(anyhow::Error),
+    RetryableError(anyhow::Error),
+}
+
+impl AppResult {
+    fn exit_code(&self) -> ExitCode {
+        match self {
+            AppResult::Success() => ExitCode::SUCCESS,
+            AppResult::UnretryableError(_) => ExitCode::FAILURE,
+            AppResult::RetryableError(_) => ExitCode::from(2),
+        }
+    }
+}
+
+impl fmt::Display for AppResult {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            AppResult::Success() => write!(f, "Success"),
+            AppResult::UnretryableError(error) => write!(f, "Error(Unretryable): {error:?}"),
+            AppResult::RetryableError(error) => write!(f, "Error(Retryable): {error:?}"),
```

Comment:
Thinking about it again there's a confusion here (and/or with the devnet error name), looking at a retryable error in the CI output this:
```
 Error(Retryable): Mithril End to End test failed

Caused by:
    Unrecoverable devnet error: `Transfer funds on chain exited with status code: 1`
``` 
So there's the message that say "we can retry this" and the cause that say "but it's unrecoverable", maybe we can find a better naming here.

Reply:
I aligned on `Retryable` naming to avoid confusion